app/assets/javascripts/common/tags.js

```diff
@@ -114,13 +124,23 @@
     return $('<button>', {
       id: 'tags-cancel-button',
       text: 'cancel',
-      click: function(){
+      click: function(e){
+	e.preventDefault();
         // restore state
-        $(TAGS.divs.container).html(TAGS.cache.html);
-        TAGS.current = TAGS.cache.tags;
-        $('#tags-save-button').remove();
-        $('#tags-cancel-button').remove();
-        $(TAGS.divs.edit).show();
+	TAGS.current = TAGS.cache.tags;
+	
+	if (TAGS.alwaysEdit) {
+	  t.render();  // in perpetual edit mode we only need to re-render
+	} else {
+	  // in the normal operation we will restore the original view
```

Comment:
perhaps we could wrap this logic in a `restore()` helper method? would group and self-document a bit. perhaps cut down on the need for explanation.